src/Directory.BeforeCommon.targets

```
@@ -117,4 +117,8 @@
     <FeatureMSIORedist>true</FeatureMSIORedist>
   </PropertyGroup>
 
+   <PropertyGroup Condition="'$(TargetFramework)' == 'net5.0'">
```

This kind of condition is a bit of a gotcha. The APIs were added in .NET 5.0, but we don't actually target that--we now target .NET 7.0. So this won't be defined! Instead you can use a compatibility check:

```suggestion
   <PropertyGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net5.0'))">
```

But we already have a few conditions like this, so I'd put it in the `'$(TargetFramework)' == 'netcoreapp2.1' or '$(TargetFramework)' == '$(LatestDotNetCoreForMSBuild)'` block above.

---

So what you are saying here is the net5.0 property group can be replaced entirely with `<PropertyGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net5.0'))">`

---

I have made the following changes and added a test to compensate for the addition in the Constants file. Let me know your thoughts.